server/src/main/java/io/crate/role/PrivilegesModifier.java

```diff
@@ -159,14 +159,14 @@ public static long dropTableOrViewPrivileges(Metadata.Builder mdBuilder,
             for (Privilege privilege : role.privileges()) {
                 Subject subject = privilege.subject();
                 Securable securable = subject.securable();
-                if (securable != Securable.TABLE && securable != Securable.VIEW) {
-                    continue;
-                }
-
-                String ident = subject.ident();
-                assert ident != null : "ident must not be null for securable 'TABLE'";
-                if (ident.equals(tableOrViewIdent)) {
-                    affectedPrivileges++;
+                if (securable == Securable.TABLE || securable == Securable.VIEW) {
+                    String ident = subject.ident();
+                    assert ident != null : "ident must not be null for securable 'TABLE' or 'VIEW'";
+                    if (ident.equals(tableOrViewIdent)) {
+                        affectedPrivileges++;
+                    } else {
+                        updatedPrivileges.add(privilege);
+                    }
```

Comment:
could be simplified, but maybe personal perference:
```suggestion
                String ident = subject.ident();
                if ((securable == Securable.TABLE || securable == Securable.VIEW)
                    && tableOrViewIdent.equals(ident)) {
                    affectedPrivileges++;
```